packages/dataviews-client/src/resolve-dataviews.ts

```diff
@@ -110,32 +104,32 @@ const getTrackDataviewDatasetConfigs = (
     (datasetConfig) => datasetConfig.query?.find((q) => q.id === 'vessels')?.value
   ) // Loitering
 
-  return {
-    info,
-    track,
-    events,
-  }
+  return [info, track, ...events]
 }
 
+export type DatasetConfigsTransforms = Partial<
+  Record<Generators.Type, (datasetConfigs: DataviewDatasetConfig[]) => DataviewDatasetConfig[]>
+>
+
 export const getDataviewsForResourceQuerying = (
   dataviewInstances: UrlDataviewInstance[],
-  trackDatasetConfigsTransform?: (
-    trackDatasetConfigs: TrackDatasetConfigs
-  ) => DataviewDatasetConfig[]
+  datasetConfigsTransform?: DatasetConfigsTransforms
 ) => {
   if (!dataviewInstances) return []
   const preparedDataviewsInstances = dataviewInstances.map((dataviewInstance) => {
-    if (dataviewInstance.config?.type !== Generators.Type.Track) return dataviewInstance
-
-    const { track, info, events } = getTrackDataviewDatasetConfigs(dataviewInstance)
     let preparedDatasetConfigs
+    switch (dataviewInstance.config?.type) {
+      case Generators.Type.Track:
+        preparedDatasetConfigs = getTrackDataviewDatasetConfigs(dataviewInstance)
+        if (datasetConfigsTransform && datasetConfigsTransform[Generators.Type.Track]) {
+          preparedDatasetConfigs =
+            datasetConfigsTransform[Generators.Type.Track]!(preparedDatasetConfigs)
```

Comment:
Isn't Eslint complaining about using [Forbidden non-null assertion](https://github.com/typescript-eslint/typescript-eslint/blob/v4.29.1/packages/eslint-plugin/docs/rules/no-non-null-assertion.md) ?
why not using `?.` instead ? what is the difference ?